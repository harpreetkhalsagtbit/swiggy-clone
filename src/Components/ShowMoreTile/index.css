.show-more-tile {
    position: relative;
    width: 254px;
    background-color: #fff;
    flex: 1 0 30%; /* explanation below */

    display: flex;
    justify-content: center;
    align-items: center;
    border:1px solid #e46d47;
}

.show-more-title {
    font-size: 24px;
    color:#e46d47;
    cursor: pointer;
    font-weight: 600;
}

