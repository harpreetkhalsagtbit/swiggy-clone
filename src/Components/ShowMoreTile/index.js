import React from "react";

import "./index.css";

const ShowMoreTile = ({ remainingItems, showMore }) => {
    // Lift state up
    let showMorehandler = () => {
        showMore()
    }
    return (
        <div className="show-more-tile">
            <div className="show-more-title" onClick={showMorehandler}>
                <span href="">+{remainingItems} More</span>
            </div>
        </div>
    )
};
export default ShowMoreTile;
