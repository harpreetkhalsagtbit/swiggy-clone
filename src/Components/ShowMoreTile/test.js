import React from 'react'
import {render, fireEvent} from '@testing-library/react'
import ShowMoreTile from '../ShowMoreTile/index.js'

describe('<ShowMoreTile /> spec', () => {
    it('renders the component', () => {
        let container = render(<ShowMoreTile/>)
        expect(container.firstChild).toMatchSnapshot()
    })
    it('assert there are 12 Show More Tiles', () => {
        render(<ShowMoreTile remainingItems="12"/>)
 
        expect(document.querySelectorAll('.show-more-title span').length).toBe(1)
        expect(document.querySelector('.show-more-title span').innerHTML).toBe("+12 More")
    })
})
