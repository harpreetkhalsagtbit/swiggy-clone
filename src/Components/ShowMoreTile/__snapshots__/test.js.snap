// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowMoreTile /> spec renders the component 1`] = `undefined`;
